frontend/src/types/schemaSystem.ts

```diff
@@ -0,0 +1,280 @@
+import { SchemaReviewPolicyId } from "./id";
+import { Principal } from "./principal";
+
+export enum RuleLevel {
+  DISABLED = "DISABLED",
+  ERROR = "ERROR",
+  WARNING = "WARNING",
+}
+
+export const LEVEL_LIST = [
+  RuleLevel.ERROR,
+  RuleLevel.WARNING,
+  RuleLevel.DISABLED,
+];
+
+export enum PayloadType {
+  STRING = "STRING",
+  STRING_ARRAY = "STRING[]",
+  TEMPLATE = "TEMPLATE",
+}
+
+interface StringPayload {
+  type: PayloadType.STRING;
+  default: string;
+  value?: string;
+}
+
+interface StringArrayPayload {
+  type: PayloadType.STRING_ARRAY;
+  default: string[];
+  value?: string[];
+}
+
+interface TemplatePayload {
+  type: PayloadType.TEMPLATE;
+  default: string;
+  templateList: { id: string; description?: string }[];
+  value?: string;
+}
+
+export interface RulePayload {
+  [key: string]: StringPayload | StringArrayPayload | TemplatePayload;
+}
+
+export type SchemaRuleEngineType = "MYSQL" | "COMMON";
+
+export type CategoryType = "ENGINE" | "NAMING" | "QUERY" | "TABLE" | "COLUMN";
+
+export interface SchemaRule {
+  id: string;
+  category: CategoryType;
+  engine: SchemaRuleEngineType;
+  description: string;
+  payload?: RulePayload;
+}
+
+export interface SelectedRule extends SchemaRule {
+  level: RuleLevel;
+}
+
+interface DatabaseSchemaRule {
+  id: string;
+  level: RuleLevel;
+  payload?: {
+    [key: string]: string | string[] | undefined;
```

Comment:
can we declare the full type:  payload? : PayloadType1 | PayloadType2

PayloadType1 {
 pk: string;
}